> If I understand correctly, this pr tries to change Failed core proc(s): {} to Failed core proc(s): {killed by SIGSEGV (-11)} right?

No, what this PR change is to show `result.stdout/result.stderr` from the subprocess

## Before this PR

result.stdout/result.stderr are swallowed since the current code captured it but didn't show 

```
models/multimodal/generation/test_whisper.py::test_models_distributed[5-200-half-ray-openai/whisper-large-v3-turbo] Fork a new process to run a test 44035
Fork a new process to run a test 0
FAILED

[~46s of silence — no engine output captured]

____ test_models_distributed[5-200-half-ray-openai/whisper-large-v3-turbo] _____
...
E   RuntimeError: Test subprocess 'test_models_distributed' failed (exit code 1):
E   Traceback (most recent call last):
E     File ".../vllm/v1/engine/utils.py", line 1183, in wait_for_engine_startup
E       raise RuntimeError(
E   RuntimeError: Engine core initialization failed. See root cause above. Failed core proc(s): {}
```

## After this PR

Could observe the stdout/stderr in the log

```
models/multimodal/generation/test_whisper.py::test_models_distributed[5-200-half-ray-openai/whisper-large-v3-turbo] Fork a new process to run a test 44035
Fork a new process to run a test 0
INFO 05-07 00:30:02 [utils.py:240] non-default args: {'dtype': 'half', 'max_model_len': 448, 'distributed_executor_backend': 'ray', 'tensor_parallel_size': 2, ...}
INFO 05-07 00:30:05 [model.py:563] Resolved architecture: WhisperForConditionalGeneration
(EngineCore pid=44120) INFO 05-07 00:30:08 [core.py:109] Initializing a V1 LLM engine ...
(EngineCore pid=44120) (RayWorkerProc pid=44210) (Worker_TP0 pid=44210) INFO ... Starting to load model openai/whisper-large-v3-turbo...
(EngineCore pid=44120) (RayWorkerProc pid=44210) (Worker_TP0 pid=44210) INFO ... Model loading took 1.55 GiB memory and 4.7 seconds
(EngineCore pid=44120) (RayWorkerProc pid=44210) (Worker_TP0 pid=44210) INFO ... Capturing CUDA graphs (decode-only, FULL_DECODE_ONLY): 0% 0/8
(EngineCore pid=44120) (RayWorkerProc pid=44210) (Worker_TP0 pid=44210) ERROR ... torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 624.00 MiB. GPU 0 has a total capacity of 22.06 GiB of which 412.50 MiB is free.
(EngineCore pid=44120) (RayWorkerProc pid=44210) (Worker_TP0 pid=44210) Traceback (most recent call last):
(EngineCore pid=44120) (RayWorkerProc pid=44210) (Worker_TP0 pid=44210)   File ".../vllm/v1/worker/gpu_model_runner.py", line 6210, in capture_model
(EngineCore pid=44120) (RayWorkerProc pid=44210) (Worker_TP0 pid=44210)     self._capture_cudagraphs(...)
(EngineCore pid=44120) ERROR 05-07 00:30:43 [core.py:807] EngineCore failed to start.
(EngineCore pid=44120) Process EngineCore_DP0:
(EngineCore pid=44120) Traceback (most recent call last):
(EngineCore pid=44120)   File ".../vllm/v1/engine/core.py", line 800, in run_engine_core
(EngineCore pid=44120)     engine_core = EngineCoreProc(*args, **kwargs)
(EngineCore pid=44120)   File ".../vllm/v1/engine/core.py", line 524, in __init__
(EngineCore pid=44120)     super().__init__(...)
(EngineCore pid=44120) torch.OutOfMemoryError: CUDA out of memory. ...
FAILED

____ test_models_distributed[5-200-half-ray-openai/whisper-large-v3-turbo] _____
...
E   RuntimeError: Engine core initialization failed. ... Failed core proc(s): {}
```

The `Failed core proc(s): {}` line stays the same; 
The `killed by SIGSEGV (-11)` formatting is for the outer test wrapper's `RuntimeError("Test subprocess ... failed (exit code N)")` and only appears if the test wrapper process itself dies from a signal. I could remove it if you think it is unnecessary.

--- a/tests/utils.py
+++ b/tests/utils.py
@@ -1511,6 +1511,16 @@ def fork_new_process_for_each_test(func: Callable[_P, None]) -> Callable[_P, Non
     return wrapper
 
 
+def _format_subprocess_exit(returncode: int) -> str:
+    """Render a subprocess exit code, naming the signal for negative codes."""
+    if returncode >= 0:
+        return f"exit code {returncode}"
+    try:
+        return f"killed by {signal.Signals(-returncode).name} ({returncode})"
+    except ValueError:
+        return f"exit code {returncode}"
+
+
 # Set on the spawn-child interpreter so the wrapper short-circuits when the
 # child resolves `module.qualname` back to its own decorated form, instead of
 # launching another subprocess.
@@ -1531,6 +1541,12 @@ def spawn_new_process_for_each_test(f: Callable[_P, None]) -> Callable[_P, None]
     ``vllm.compilation.counter.compilation_counter``) into stale clones in
     the child, so increments performed by the production code in the child
     would never be observable to the test.
+
+    The child inherits the parent's stdout/stderr so its output (engine
+    cores, NCCL, CUDA, ...) reaches the test runner live; the Python-level
+    traceback is serialized to ``tb_file`` for structured re-raising. A
+    native crash leaves ``tb_file`` empty — the diagnostic is then only in
+    the inherited subprocess output.
     """
 
     @functools.wraps(f)
@@ -1581,23 +1597,20 @@ def spawn_new_process_for_each_test(f: Callable[_P, None]) -> Callable[_P, None]
             result = subprocess.run(
                 [sys.executable, "-c", child_script],
                 input=payload,
-                capture_output=True,
                 env=env,
             )
 
             if result.returncode != 0:
-                # Prefer the child's traceback file; fall back to stderr if
-                # the child crashed before its except handler ran.
                 try:
                     with open(tb_file) as fp:
                         tb = fp.read()
                 except OSError:
                     tb = ""
                 if not tb:
-                    tb = result.stderr.decode()
+                    tb = "<no Python traceback; see subprocess output above>"
                 raise RuntimeError(
                     f"Test subprocess '{f.__name__}' failed "
-                    f"(exit code {result.returncode}):\n{tb}"
+                    f"({_format_subprocess_exit(result.returncode)}):\n{tb}"
                 )
         finally:
             with contextlib.suppress(OSError):

